src/geometry/LineUtil.js

```diff
@@ -240,3 +241,59 @@ export function _flat(latlngs) {
 	console.warn('Deprecated use of _flat, please use L.LineUtil.isFlat instead.');
 	return isFlat(latlngs);
 }
+
+/* @function polylineCenter(latlngs: LatLng[], map: Map): LatLng
+ * Returns the center of the `Polyline` of the passed LatLngs.
+ * The returned LatLng will be on the `Polyline`.
+ */
+export function polylineCenter(latlngs, map) {
+	var i, halfDist, segDist, dist, p1, p2, ratio, center;
+
+	if (!latlngs || latlngs.length === 0) {
+		throw new Error('latlngs not passed');
+	}
+
+	if (!isFlat(latlngs)) {
+		throw new Error('latlngs are not flat!');
+	}
+
+	// throws error when map is not passed because calculation requires projected coordinates
+	if (!map) {
+		throw new Error('map not passed');
+	}
+
+	var zoom = map.getMaxZoom() === Infinity ? map.getZoom() : map.getMaxZoom();
+	var points = [];
+	for (var j in latlngs) {
+		points.push(map.project(toLatLng(latlngs[j]), zoom));
+	}
+	var len = points.length;

```

Comment:
Isn't that the same value as `latlngs.length`?

Reply:
I copied this from the original code. But I think it is done because `latlngs.length` returns not the correct length if it is a multipolyline.

Reply:
> I copied this from the original code.

But it is redundant for the new code, right?

Reply:
Yep if 
> But I think it was / is used because `latlngs.length` returns not the correct length if it is a multipolyline.

makes no problems

Reply:
But here latlngs mapped to points directly, so I do not see any difference.

Reply:
I need to check this mapping because this is not working for mult-lines. Currently `latlngs.length` work because I check with `if (!isFlat(latlngs)) {` if it is a multi-line. If we say that multipolylines should supported too, it will not work.

Can it be that the old `getCenter()` is also only working for **no**t mulit-lines? Because it uses only the first array `points = this._rings[0],`